### PR TITLE
Fix destination host validataion

### PR DESCRIPTION
**Pull request by frankbu**

### CODE REVIEW

**Comment by linsun** [comment]
fixes: #12794

File: pilot/pkg/model/validation_test.go [hunk 1]
```diff
@@ -2261,6 +2261,21 @@ func TestValidateRouteDestination(t *testing.T) {
...
+			Destination: &networking.Destination{Host: "*foo.baz"},
+		}}, valid: true},
+		{name: "wildcard", routes: []*networking.RouteDestination{&networking.RouteDestination{
+			Destination: &networking.Destination{Host: "*"},
+		}}, valid: true},
+		{name: "bad wildcard", routes: []*networking.RouteDestination{&networking.RouteDestination{
+			Destination: &networking.Destination{Host: "foo.*"},
+		}}, valid: false},
+		{name: "bad fqdn", routes: []*networking.RouteDestination{&networking.RouteDestination{
+			Destination: &networking.Destination{Host: "default/baz"},
```
Comment (linsun) [inline pilot/pkg/model/validation_test.go]: use FQDN in the test to match the name "bad fqdn"?

File: pilot/pkg/model/validation.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ const (
 	dns1123LabelMaxLength int    = 63
 	dns1123LabelFmt       string = "[a-zA-Z0-9]([-a-z-A-Z0-9]*[a-zA-Z0-9])?"
 	// a wild-card prefix is an '*', a normal DNS1123 label with a leading '*' or '*-', or a normal DNS1123 label
-	wildcardPrefix string = `\*|(\*|\*-)?(` + dns1123LabelFmt + `)`
+	wildcardPrefix string = `(\*|(\*|\*-)?` + dns1123LabelFmt + `)`
```
Comment (linsun) [inline pilot/pkg/model/validation.go]: It is not clear how this change fixes things... as I don't see us not allow `/` here.

File: pilot/pkg/model/validation_test.go (hunk 1 above)
Comment (frankbu) [reply pilot/pkg/model/validation_test.go]: `/` is not a valid char in a FQDN.

File: pilot/pkg/model/validation.go (hunk 2 above)
Comment (frankbu) [reply pilot/pkg/model/validation.go]: `(` was in wrong place, so the regex didn't work as expected

Comment (linsun) [reply pilot/pkg/model/validation.go]: Thanks, I missed wildcardPrefixRegexp() line 65.

File: pilot/pkg/model/validation_test.go (hunk 1 above)
Comment (linsun) [reply pilot/pkg/model/validation_test.go]: yes, I was referring to something like default/baz.default.svc.cluster.local. :)

**Pull request merged**